File: Data Structures & Algorithms/first-missing-positive/submission-5.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)

        for i in range(n):
            if nums[i] <= 0 or nums[i] > n:
                nums[i] = n + 1

        for i in range(n):
            id = abs(nums[i])

            if id > n:
                continue
            id -= 1

            if nums[id] > 0:
                nums[id] = -nums[id]

        for i in range(n):
            if nums[i] > 0:
                return i + 1
        return n + 1

File: Data Structures & Algorithms/sliding-window-maximum/submission-5.java
public class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        PriorityQueue<int[]> heap = new PriorityQueue<>((a, b) -> b[0] - a[0]);
        int[] output = new int[nums.length - k + 1];
        int idx = 0;
        for (int i = 0; i < nums.length; i++) {
            heap.offer(new int[]{nums[i], i});
            if (i >= k - 1) {
                while (heap.peek()[1] <= i - k) {
                    heap.poll();
                }
                output[idx++] = heap.peek()[0];
            }
        }
        return output;
    }
}